### PR TITLE
create endpoints resource

### PR DESCRIPTION
**Pull request by dbasunag**

### CODE REVIEW

File: ocp_resources/endpoints.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from ocp_resources.resource import NamespacedResource
+
+
+class Endpoints(NamespacedResource):
+    """
+    Endpoints object. API reference:
+    https://docs.openshift.com/container-platform/4.12/rest_api/network_apis/endpoints-v1.html#endpoints-v1
+    """
+
+    api_version = "v1"
```
Comment (rnetser) [inline ocp_resources/endpoints.py:11]:
```suggestion
    api_version = Resource.ApiVersion.V1
```

File: ocp_resources/endpoints.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            yaml_file=yaml_file,
+            delete_timeout=delete_timeout,
+            **kwargs,
+        )
+        self.addresses = addresses
+        self.ports = ports
+
+    def to_dict(self):
+        super().to_dict()
+        if not self.yaml_file:
```
Comment (rnetser) [inline ocp_resources/endpoints.py:41]: please raise if `addresses` or `ports` are not provided.

File: ocp_resources/endpoints.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        namespace=None,
+        client=None,
+        addresses=None,
+        ports=None,
+        teardown=True,
+        privileged_client=None,
+        yaml_file=None,
+        delete_timeout=TIMEOUT_4MINUTES,
+        **kwargs,
+    ):
```
Comment (rnetser) [inline ocp_resources/endpoints.py:25]: please add docstring (mainly for ports and addresses)

**Review by rnetser — CHANGES_REQUESTED** [review]

File: ocp_resources/endpoints.py (hunk 3 above)
Comment (dbasunag) [reply ocp_resources/endpoints.py:25]: Added.

File: ocp_resources/endpoints.py (hunk 2 above)
Comment (dbasunag) [reply ocp_resources/endpoints.py:53]: Done.

**Comment by dbasunag** [comment]
!verified

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Comment by dbasunag** [comment]
!cherry-pick v4.12

**Pull request merged**